assets/js/controllers/settings/addresses.controller.js

```diff
@@ -2,75 +2,86 @@ angular
   .module('walletApp')
   .controller('SettingsAddressesCtrl', SettingsAddressesCtrl);
 
-function SettingsAddressesCtrl ($scope, $state, $stateParams, $q, $sce, Wallet, MyWalletHelpers, MyBlockchainApi, Alerts, paymentRequests, $uibModal) {
-  $scope.paymentRequests = paymentRequests;
+function SettingsAddressesCtrl ($scope, $translate, $rootScope, $state, $stateParams, $q, $sce, Wallet, Labels, MyWalletHelpers, MyBlockchainApi, Alerts, $uibModal) {
+  if (!Wallet.status.isLoggedIn) {
+    console.error('Controller depends on being logged in');
+    return;
+  }
+
+  let accountIndex = parseInt($stateParams.account, 10);
+
+  $scope.presentFilter = (address) => address && !!address.label && address.used === false;
+
+  $scope.pastAddressesPage = (page, pageLength) => {
+    let pastFilter = (address) => address && (!address.label || address.used === true);
+
+    return $scope.addresses.filter(pastFilter)
+                           .reverse().slice((page - 1) * pageLength, page * pageLength);
+  };
```

Comment:
Also, following these steps from @jtormey to create a "used" address I get an error:

Steps: Label a new address -> Send to current receive address / Label another address -> Remove label from address in step 1

Error: [ngRepeat:dupes] Duplicates in a repeater are not allowed. Use 'track by' expression to specify unique keys. Repeater: address in addresses | filter:presentFilter | filter:searchText | orderBy:'-', Duplicate key: object:1227, Duplicate value: {"label":"New Address"}